FAIMS/FAIMS3
# feat: custom hook for offline/online status and fallback component

# feat: custom hook for offline/online status and fallback component

## JIRA Ticket

[BSS-321](https://jira.csiro.com/browse/BSS-321)

## Description

Adds a custom hook for offline/online status and fallback component which can be rendered to display connectivity issues. 

Also adds example component which can be added to pages to test functionality. Will need to remove before merging.

## How to Test

Currently the home page will show the offline/online as a test component. You can then use the browser dev mode to simulate different network conditions and make sure the component responds as intended. 

![image](https://github.com/user-attachments/assets/5e6e0937-b04e-49c5-8a96-a4a78f3f2612)

This is the fallback component - note that refresh just checks the online status again with window navigator, and return home goes to index. The refresh button will always show a short load to reassure user. 

@luke-mcfarlane-rocketlab  @stevecassidy  if you could test this works similarly on the android app that would be great.

## Additional Information

Currently can't get the useStyles to pickup the theme properly. That's why I have marked this as draft. I'd like to refactor the fallback styling to use the theme palette etc once this is working.

## Checklist
